crates/connectors/ndc-postgres/src/schema.rs

```diff
@@ -261,3 +271,43 @@ fn type_to_type(typ: &metadata::Type) -> models::Type {
         },
     }
 }
+
+fn mutation_to_procedure(name: &String, mutation: &generate::Mutation) -> models::ProcedureInfo {
+    match mutation {
+        generate::Mutation::DeleteMutation(delete) => delete_to_procedure(name, delete),
+    }
+}
+
+fn delete_to_procedure(name: &String, delete: &delete::DeleteMutation) -> models::ProcedureInfo {
+    match delete {
+        delete::DeleteMutation::DeleteByKey {
+            by_column,
+            description,
+            table_name: ast::TableName(table_name),
+            schema_name: ast::SchemaName(schema_name),
+        } => {
+            let mut arguments = BTreeMap::new();
+
+            let result_type = if schema_name == "public" {
```

Comment:
I dislike this, there must be a nicer way to get a matching type name for a collection.

Reply:
Oh yeah. Which schemas are qualified or not depends on ndc raw config too.

The logic that decides which names are used for collection types should really be centralized so that we use the same everywhere we need that.

Reply:
Ahh, so it turns out the introspection itself is the source of this logic, and uses it to generate the keys in the `metadata.tables` map, so using those instead of generating things.